src/utils/api/helper.ts

```diff
@@ -262,6 +262,29 @@ const activePaymentTask = (
               let reason;
               if (responseType.status === 200) {
                 reason = "";
+                const cartInfo = getSessionItem(SessionItems.cart) as
+                  | Cart
+                  | undefined;
+                if (cartInfo !== undefined) {
+                  const updatedPaymentNotices = cartInfo.paymentNotices.map(
+                    (paymentNotice) => {
+                      const updatedAmount = responseType.value.payments.find(
+                        (p) =>
+                          p.rptId ===
+                          (`${paymentNotice.fiscalCode}${paymentNotice.noticeNumber}` as RptId)
```

Comment:
Is this cast ok? Isn't `RptId` an `io-ts` tagged type?

Reply:
I hope it is, we use it several times in this PR

Reply:
Just checked, _as a type_ `RptId` is an alias of `string & PatternedString<Pattern>`, so can be safely coerced with `as`